pkger/clone_resource.go

```diff
@@ -220,77 +217,255 @@ func (ex *resourceExporter) resourceCloneToKind(ctx context.Context, r ResourceT
-		ch, err := ex.checkSVC.FindCheckByID(ctx, r.ID)
+		if n < 1 {
+			return errors.New("no buckets found")
+		}
+
+		for _, bkt := range bkts {
+			mapResource(bkt.OrgID, bkt.ID, KindBucket, BucketToObject(r.Name, *bkt))
+		}
+	case r.Kind.is(KindCheck), r.Kind.is(KindCheckDeadman), r.Kind.is(KindCheckThreshold):
+		filter := influxdb.CheckFilter{}
+		if r.ID != influxdb.ID(0) {
+			filter.ID = &r.ID
+		}
+		if len(r.Name) > 0 {
+			filter.Name = &r.Name
+		}
+		chs, n, err := ex.checkSVC.FindChecks(ctx, filter)
 		if err != nil {
 			return err
 		}
-		mapResource(ch.GetOrgID(), uniqByNameResID, KindCheck, CheckToObject(r.Name, ch))
+		if n < 1 {
+			return errors.New("no checks found")
+		}
+
+		for _, ch := range chs {
+			mapResource(ch.GetOrgID(), ch.GetID(), KindCheck, CheckToObject(r.Name, ch))
+		}
 	case r.Kind.is(KindDashboard):
-		dash, err := findDashboardByIDFull(ctx, ex.dashSVC, r.ID)
+		var (
+			hasID  bool
+			filter = influxdb.DashboardFilter{}
+		)
+		if r.ID != influxdb.ID(0) {
+			hasID = true
+			filter.IDs = []*influxdb.ID{&r.ID}
+		}
+
+		dashs, _, err := ex.dashSVC.FindDashboards(ctx, filter, influxdb.DefaultDashboardFindOptions)
 		if err != nil {
 			return err
 		}
-		mapResource(dash.OrganizationID, dash.ID, KindDashboard, DashboardToObject(r.Name, *dash))
+
+		var mapped bool
+		for _, dash := range dashs {
+			if (!hasID && len(r.Name) > 0 && dash.Name != r.Name) || (hasID && dash.ID != r.ID) {
+				continue
+			}
+
+			for _, cell := range dash.Cells {
+				v, err := ex.dashSVC.GetDashboardCellView(ctx, dash.ID, cell.ID)
+				if err != nil {
+					continue
+				}
+				cell.View = v
+			}
+
+			mapResource(dash.OrganizationID, dash.ID, KindDashboard, DashboardToObject(r.Name, *dash))
+			mapped = true
+		}
+
+		if !mapped {
+			return errors.New("no dashboards found")
+		}
 	case r.Kind.is(KindLabel):
-		l, err := ex.labelSVC.FindLabelByID(ctx, r.ID)
-		if err != nil {
-			return err
+		switch {
+		case r.ID != influxdb.ID(0):
+			l, err := ex.labelSVC.FindLabelByID(ctx, r.ID)
+			if err != nil {
+				return err
+			}
+
+			mapResource(l.OrgID, uniqByNameResID, KindLabel, LabelToObject(r.Name, *l))
+		case len(r.Name) > 0:
+			labels, err := ex.labelSVC.FindLabels(ctx, influxdb.LabelFilter{Name: r.Name})
+			if err != nil {
+				return err
+			}
+
+			for _, l := range labels {
+				mapResource(l.OrgID, uniqByNameResID, KindLabel, LabelToObject(r.Name, *l))
+			}
 		}
-		mapResource(l.OrgID, uniqByNameResID, KindLabel, LabelToObject(r.Name, *l))
 	case r.Kind.is(KindNotificationEndpoint),
 		r.Kind.is(KindNotificationEndpointHTTP),
 		r.Kind.is(KindNotificationEndpointPagerDuty),
 		r.Kind.is(KindNotificationEndpointSlack):
-		e, err := ex.endpointSVC.FindNotificationEndpointByID(ctx, r.ID)
-		if err != nil {
-			return err
+		var endpoints []influxdb.NotificationEndpoint
+
+		switch {
+		case r.ID != influxdb.ID(0):
+			ndpoint, err := ex.endpointSVC.FindNotificationEndpointByID(ctx, r.ID)
```

Comment:
should this be `endpoint`?

can it be called `notifEndpoint` if `endpoint` is taken?